stdlib/Artifacts/src/Artifacts.jl

```diff
@@ -0,0 +1,634 @@
+module Artifacts
+
+import Base: get, SHA1
+using Base.BinaryPlatforms, Base.TOML
+
+export artifact_exists, artifact_path, remove_artifact, verify_artifact,
+       artifact_meta, artifact_hash, find_artifacts_toml, @artifact_str
+
+"""
+    parse_toml(path::String)
+
+Uses Base.TOML to parse a TOML file
+"""
+function parse_toml(path::String)
+    p = Base.TOML.Parser()
+    Base.TOML.reinit!(p, read(path, String); filepath=path)
+    return Base.TOML.parse(p)
+end
+
+# keep in sync with Base.project_names and Base.manifest_names
+const artifact_names = ("JuliaArtifacts.toml", "Artifacts.toml")
+
+const ARTIFACTS_DIR_OVERRIDE = Ref{Union{String,Nothing}}(nothing)
+"""
+    with_artifacts_directory(f::Function, artifacts_dir::String)
+
+Helper function to allow temporarily changing the artifact installation and search
+directory.  When this is set, no other directory will be searched for artifacts, and new
+artifacts will be installed within this directory.  Similarly, removing an artifact will
+only effect the given artifact directory.  To layer artifact installation locations, use
+the typical Julia depot path mechanism.
+"""
+function with_artifacts_directory(f::Function, artifacts_dir::String)
+    try
+        ARTIFACTS_DIR_OVERRIDE[] = artifacts_dir
+        f()
+    finally
+        ARTIFACTS_DIR_OVERRIDE[] = nothing
+    end
+end
+
+"""
+    artifacts_dirs(args...)
+
+Return a list of paths joined into all possible artifacts directories, as dictated by the
+current set of depot paths and the current artifact directory override via the method
+`with_artifacts_dir()`.
+"""
+function artifacts_dirs(args...)
+    if ARTIFACTS_DIR_OVERRIDE[] === nothing
+        return [abspath(depot, "artifacts", args...) for depot in Base.DEPOT_PATH]
+    else
+        # If we've been given an override, use _only_ that directory.
+        return [abspath(ARTIFACTS_DIR_OVERRIDE[], args...)]
+    end
+end
+
+"""
+    ARTIFACT_OVERRIDES
+
+Artifact locations can be overridden by writing `Override.toml` files within the artifact
+directories of Pkg depots.  For example, in the default depot `~/.julia`, one may create
+a `~/.julia/artifacts/Override.toml` file with the following contents:
+
+    78f35e74ff113f02274ce60dab6e92b4546ef806 = "/path/to/replacement"
+    c76f8cda85f83a06d17de6c57aabf9e294eb2537 = "fb886e813a4aed4147d5979fcdf27457d20aa35d"
+
+    [d57dbccd-ca19-4d82-b9b8-9d660942965b]
+    c_simple = "/path/to/c_simple_dir"
+    libfoo = "fb886e813a4aed4147d5979fcdf27457d20aa35d""
+
+This file defines four overrides; two which override specific artifacts identified
```

Comment:
Referring to "this file" in a docstring is a bit non-ideal maybe. The function could be moved and the docstring (as opposed to a comment) is mostly supposed to be read in some other medium like in the terminal or html rendering where the concept of the file is just an implementation detail.

Reply:
It seems very straightforward to me what file it's referring to: the ``~/.julia/artifacts/Override.toml` file it was talking about in the paragraph before.

Reply:
I understand now.